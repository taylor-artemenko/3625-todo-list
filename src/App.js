import React, { useReducer, createContext } from "react";
import { Route, Switch } from 'react-router-dom';
import { createGlobalStyle, ThemeProvider } from "styled-components";
import { reducer, initialState } from "./reducer";

import Home from './components/Home';
import About from './components/About';
import Navbar from './components/Navbar';
import Error from './components/Error';
import Api from './components/Api';

export const AppContext = createContext();

function App() {
    const [state, dispatch] = useReducer(reducer, initialState);
    const { currentTheme } = state;

    return (
        <ThemeProvider theme={currentTheme}>
            <AppContext.Provider value={{ ...state, dispatch }}>
                <GlobalStyles />
                <div className='todoapp stack-large'>
                    <Navbar />
                    <Switch>
                        <Route path='/' component={Home} exact />
                        <Route path='/about' component = {About} />
                        <Route path='/api' component={Api} />
                        <Route component={Error} />
                    </Switch>
                </div>
            </AppContext.Provider>
        </ThemeProvider>
    );
}

const GlobalStyles = createGlobalStyle`
  html, body, .todoapp, a, button {
    background: ${props => props.theme.backgroundColor};
    color: ${props => props.theme.textColor};
    font-family: sans-serif;
  }
`;

export default App