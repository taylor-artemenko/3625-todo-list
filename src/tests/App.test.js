import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import App from "../App";

describe("App.js tests", function () {
    const empty_tasklist = [];
    const mock_tasks =
        [{ id: "todo-0", name: "task1", completed: true },
        { id: "todo-1", name: "task2", completed: false },
        { id: "todo-2", name: "task3", completed: false }]

    test("Test verifies the App component renders without crashing", function () {
        const div = document.createElement("div");
        ReactDOM.render(<App tasks={empty_tasklist} />, div);
    });

    test("Test verifies the App component should display all todo items", function () {
        //Test with three tasks
        const component = renderer.create(<App tasks = {mock_tasks}
        />);
        const testInstance = component.toJSON();
        expect(testInstance.children[4].children.length).toBe(3);

        //Test with four tasks
        const component2 = renderer.create(<App tasks = {[
            { id: "todo-0", name: "task1", completed: true },
            { id: "todo-1", name: "task2", completed: false },
            { id: "todo-2", name: "task3", completed: false },
            { id: "todo-3", name: "task4", completed: false }
        ]}
        />);const testInstance2 = component2.toJSON();
        expect(testInstance2.children[4].children.length).toBe(4);
    });

    test('Test verifies that the heading text should always show correct number of tasks', function () {
        //Test with three tasks
        const component = renderer.create(<App tasks = {mock_tasks}
        />);
        const testInstance = component.toJSON();
        expect(testInstance.children[3].children[0]).toMatch('3 tasks remaining');

        //Test with two tasks
        const component2 = renderer.create(<App tasks = {[
            { id: "todo-0", name: "task1", completed: true },
            { id: "todo-1", name: "task2", completed: false }
        ]}
        />);
        const testInstance2 = component2.toJSON();
        expect(testInstance2.children[3].children[0]).toMatch('2 tasks remaining');
    });

    test("the component should match snapshot", function () {
        const tree = renderer.create(<App tasks={empty_tasklist} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});