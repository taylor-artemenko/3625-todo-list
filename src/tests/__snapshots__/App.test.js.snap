// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js tests the component should match snapshot 1`] = `
<div
  className="todoapp stack-large"
>
  <h1>
    TodoMatic
  </h1>
  <form
    onSubmit={[Function]}
  >
    <h2
      className="label-wrapper"
    >
      <label
        className="label__lg"
        htmlFor="new-todo-input"
      >
        What needs to be done?
      </label>
    </h2>
    <input
      autoComplete="off"
      className="input input__lg"
      id="new-todo-input"
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn btn__primary btn__lg"
      type="submit"
    >
      Add
    </button>
  </form>
  <div
    className="filters btn-group stack-exception"
  >
    <button
      aria-pressed={true}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        All
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
    <button
      aria-pressed={false}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        Active
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
    <button
      aria-pressed={false}
      className="btn toggle-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        className="visually-hidden"
      >
        Show 
      </span>
      <span>
        Completed
      </span>
      <span
        className="visually-hidden"
      >
         tasks
      </span>
    </button>
  </div>
  <h2
    id="list-heading"
  >
    0 tasks remaining
  </h2>
  <ul
    aria-labelledby="list-heading"
    className="todo-list stack-large stack-exception"
    role="list"
  />
</div>
`;
